### PR TITLE
[receiver/snowflakereceiver] add client and config

### PR DESCRIPTION
**Pull request by shalper2**
**Description:** Implementing client.go and config.go.

**Link to tracking Issue:** [14754](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14754)

**Testing:** limited unit testing as missing factory.go limits the ability to test. Sql mocking for client.go.

**Documentation:** Added documentation referring to config.go including an example config.

### CODE REVIEW

**Comment by dmitryax** [comment]
@shalper2 I'll review it in detail tomorrow. Please rebase and make sure all tests pass. I see at least one lint job is failing. Rebase is needed because we've just recovered the broken CI https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17037

File: receiver/snowflakereceiver/client.go [hunk 1]
```diff
@@ -13,3 +13,426 @@
...
+// and a logger
+type defaultSnowflakeClient struct {
+	client *sql.DB
+	dsn    *string
+	logger *zap.Logger
+}
+
+var _ snowflakeClient = (*defaultSnowflakeClient)(nil)
+
+type snowflakeClient interface {
```
Comment (dmitryax) [inline receiver/snowflakereceiver/client.go:50]: Why do we need the interface? I don't see where it's implemented other than by `defaultSnowflakeClient`. I think it'll be simpler and easier to maintain if we keep just `snowflakeClient` struct unless we plan to have any other implementations

File: receiver/snowflakereceiver/client.go [hunk 2]
```diff
@@ -13,3 +13,426 @@
...
+		return nil, err
+	}
+
+	return rows, nil
+}
+
+// these wrap readDB and return the associated data type which the scraper will
+// use to generate and emit metrics. Which of these are called will be based on which metrics
+// are enabled in the Config (default is all of them)
+func (c defaultSnowflakeClient) GetBillingMetrics(ctx context.Context) (*[]billingMetric, error) {
```
Comment (dmitryax) [inline receiver/snowflakereceiver/client.go:110]: nit: `Get` prefix is considered redundant in golang world, so maybe `billingMetrics` or `fetchBillingMetrics`.

**Review by dmitryax — COMMENTED** [review]
Overall looks pretty good. Just a couple of nits. And please rebase so we can merge it

File: receiver/snowflakereceiver/client.go (hunk 2 above)
Comment (shalper2) [reply receiver/snowflakereceiver/client.go:110]: Thanks for the note on this, I was unaware!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**